funnel/models/event.py

```diff
@@ -28,7 +28,30 @@ def make_private_key():
     )
 
 
-class Event(BaseScopedNameMixin, db.Model):
+class ScopedNameTitleMixin(object):
+    # TODO: Move this into coaster?
+    @classmethod
+    def get(cls, parent, current_name=None, current_title=None):
+        if current_name:
```

Comment:
I think you'll need to ensure both aren't provided because of the small possibility of name and title being out of sync in real world data (`make_name` drops some characters when constructing names, so it is technically possible for two titles to have the same generated name). Currently the TypeError is raised only if _neither_ is provided, not _both_.